invenio_rdm_records/records/dumpers.py

```diff
@@ -59,3 +65,46 @@ def load(self, data, record_cls):
         # `None` covers the cases where exceptions were raised in _dump
         parent_data.pop(f"{self.key}_start", None)
         parent_data.pop(f"{self.key}_end", None)
+
+
+class LocationsDumper(ElasticsearchDumperExt):
+    """ElasticSearch dumper for location data.
+
+    This does a few things:
+
+    * Removes/adds FeatureCollection and Feature types, as those are
+      constant
+    * Calculates a centroid for each location if a geometry is provided
+
+    Centroid calculation requires shapely, except for the simple case where
+    the geometry is a point.
+    """
+
+    def dump(self, record, data):
+        """Dump the data."""
+        if 'locations' not in data:
+            return data
+        data['locations'].pop('type')
```

Comment:
minor: Is it worth to set it here to `FeatureCollection`? It seems that it is always removed by dumper and set to this value by loader. Not sure if it's value has some implication on ES queries.

Reply:
The idea was that because it was always constant, we didn't need to pass it to ES (https://github.com/inveniosoftware/invenio-rdm-records/issues/265#issuecomment-731277669)

Reply:
Ouch, I think we have misunderstood each other here. What I had in mind here, was that don't store ``"type": "FeatureCollection/Feature"`` in neither the database nor Elasticsearch. Just no ``type`` in the JSONSchema nor the ES mapping. Instead we add the ``type`` in a resource serializer. Right now we only have a [UISerializer](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/resources/serializers/ui/__init__.py#L18), but it would be similar......

Perhaps easiest to discuss IRL

Reply:
I'm going to resolve this per the first bullet point in https://github.com/inveniosoftware/invenio-rdm-records/pull/292#issuecomment-734183741.